nsxt/resource_nsxt_node_user.go

```diff
@@ -0,0 +1,240 @@
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/node"
+)
+
+func resourceNsxtUsers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtNodeUserCreate,
+		Read:   resourceNsxtNodeUserRead,
+		Update: resourceNsxtNodeUserUpdate,
+		Delete: resourceNsxtNodeUserDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"full_name": {
+				Type:        schema.TypeString,
+				Description: "Full name for the user",
+				Required:    true,
+			},
+			"last_password_change": {
+				Type:        schema.TypeInt,
+				Description: "Number of days since password was last changed",
+				Computed:    true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Description: "Password for the user",
+				Sensitive:   true,
+				Optional:    true,
+			},
+			"password_change_frequency": {
+				Type:         schema.TypeInt,
+				Description:  "Number of days password is valid before it must be changed",
+				Optional:     true,
+				Default:      90,
+				ValidateFunc: validation.IntBetween(0, 9999),
+			},
+			"password_change_warning": {
+				Type:         schema.TypeInt,
+				Description:  "Number of days before user receives warning message of password expiration",
+				Optional:     true,
+				Default:      7,
+				ValidateFunc: validation.IntBetween(0, 9999),
+			},
+			"password_reset_required": {
+				Type:        schema.TypeBool,
+				Description: "Boolean value that states if a password reset is required",
+				Computed:    true,
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Description: "Boolean value that states if the user account is activated",
+				Optional:    true,
+				Default:     true,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Description: "User status",
+				Computed:    true,
+			},
+			"user_id": {
+				Type:        schema.TypeInt,
+				Description: "Numeric id for the user",
+				Computed:    true,
+			},
+			"username": {
+				Type:         schema.TypeString,
+				Description:  "User login name",
+				ValidateFunc: validateUsername(),
+				Required:     true,
+			},
+		},
+	}
+}
+
+func validateUsername() schema.SchemaValidateFunc {
+	return func(i interface{}, k string) (s []string, es []error) {
+		v, ok := i.(string)
+		if !ok {
+			es = append(es, fmt.Errorf("expected type of %s to be string", k))
+			return
+		}
+
+		r := regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9@-_.\\-]*$")
+		if ok := r.MatchString(v); !ok {
+			es = append(es, fmt.Errorf("must be a valid username matching: ^[a-zA-Z][a-zA-Z0-9@-_.\\-]*$"))
```

Comment:
I don't think we should expect users to understand regex..

Reply:
removed regex from the error message.